feat: implement parallel prompts

d-loose opened the pull request:
This PR enables the `prompting-client` to handle multiple prompts in parallel (currently one prompt per process ID, but we'll likely use the cgroup path in the future, see https://github.com/canonical/snapd/pull/15924).

UD106
UDENG-7803
UDENG-7890

gabelluardo reviewed (commented):
This is really good, thanks!

d-loose commented:
@gabelluardo thanks for the review - I've addressed your comments and switched from pids to cgroup paths. Integration tests are failing now, since the corresponding changes on the snapd side haven't been merged and published to `edge` yet.

d-loose commented on prompting-client/src/daemon/worker.rs:
```diff
@@ -236,60 +242,87 @@ where
-            _ => return Ok(()),
-        };
+    async fn process_next_pending_prompts(&mut self) -> Result<()> {
+        let mut new_prompts = HashMap::new();
```
Nice, thanks!
